src/exportMapBuilder.js

```diff
@@ -201,6 +322,19 @@ export default class ExportMapBuilder {
 
     let hasDynamicImports = false;
 
+    function remotePath(value) {
+      return resolve.relative(value, path, context.settings);
+    }
+
+    /**
+     * The creation of this closure is isolated from other scopes
+     * to avoid over-retention of unrelated variables, which has
+     * caused memory leaks. See #1266.
+     */
+    function thunkFor(p, context) {
+      return () => ExportMapBuilder.for(childContext(p, context));
+    }
```

Comment:
putting this in the class means it's redefined every time `parse` is called. I'm fine with an eslint override here rather than affecting runtime performance

Reply:
Ok.

- https://github.com/import-js/eslint-plugin-import/pull/2989/commits/4d276ef4a115c9bcc2fdc7f4d0488a2c3291b5d0